src/cliffordsynthesis/TargetMetricHandler.cpp

```diff
@@ -27,22 +27,20 @@ namespace cs {
             }
         }
     }
-    void TargetMetricHandler::makeGateMetric(const SynthesisData& data, bool onlyCNOT) {
+    void TargetMetricHandler::makeGateMetric(const SynthesisData& data, bool onlyCNOT, const std::uint16_t cnotMultiplier) {
         logicbase::LogicTerm changes = logicbase::LogicTerm(true);
         // COST
         logicbase::LogicTerm cost = logicbase::LogicTerm(0);
         for (std::size_t gateStep = 1; gateStep < data.timesteps + 1; ++gateStep) {
             for (std::size_t a = 0; a < data.nqubits; ++a) {
-                if (!onlyCNOT) {
-                    for (auto gate: Gates::SINGLE_QUBIT_WITHOUT_NOP) {
-                        cost = cost + data.gS[gateStep][Gates::toIndex(gate)][a];
-                    }
+                for (auto gate: Gates::SINGLE_QUBIT_WITHOUT_NOP) {
+                    cost = cost + data.gS[gateStep][Gates::toIndex(gate)][a];
                 }
                 for (std::size_t b = 0; b <= a; ++b) {
                     if (a == b) {
                         continue;
                     }
-                    cost = cost + data.gTwoQubit[gateStep][a][b] + data.gTwoQubit[gateStep][b][a];
+                    cost = cost + (data.gTwoQubit[gateStep][a][b] + data.gTwoQubit[gateStep][b][a]) * logicbase::LogicTerm(onlyCNOT ? cnotMultiplier : 1);
```

Comment:
This change does not make sense to me. If I only care for the CNOT count, then it does not matter how many single-qubit gates there are.
With this change, a circuit with `x+1` CNOT gates and `y-11` single-qubit gates would be preferred over a circuit with `x` CNOT gates and `y` single-qubit gates.
Thus, yielding a sub-optimal solution.

Please revert this, if there is no argument against the above analysis.